### ISSUE TITLE
How to match end of input?

### ISSUE THREAD

**Issue by gajus** [kach/nearley#626]
I need to distinguish between a whitespace and an empty string, i.e. `foo:`, `foo:bar` and `foo: ` (this one has space) need to be treated differently.

Is there a special keyword for end of input?

**Comment by PoolloverNathan** [comment]
No; end of input is (implicitly) matched at the end of the main rule.

**Comment by PoolloverNathan** [comment]
This repository seems dead, but if someone is still around, I would like to propose syntax:
- A `?=` token in `completeexpression` which is followed by another `expr`. The second expr will be matched at the current position as if the `?=` wasn't there, but the cursor will not be affected/will be rewinded. (It exists in my fork, but is not implemented yet.)
- A `$` token at the end of `completeexpression` which causes it to only match at the end of input.
